File: Back/src/ProEventos.Domain/Models/Palestrante.cs
namespace ProEventos.Domain.Models;

public class Palestrante
{
  public int Id { get; set; }
  public string? Nome { get; set; }

  public string? MiniCurriculo { get; set; }

  public string? ImagemURL { get; set; }

  public string? Telefone { get; set; }

  public string? Email { get; set; }

  // pode possuir VARIAS redes sociais (um pra muitos)
  public IEnumerable<RedeSocial>? RedesSociais { get; set; } 

  // Associacao da tabela auxiliar (muitos pra muitos)
  public IEnumerable<PalestranteEvento>? PalestrantesEventos { get; set; }
}

File: Front/ProEventos-App/src/app/models/Palestrante.ts
import { Evento } from "./Evento";
import { RedeSocial } from "./RedeSocial";

export interface Palestrante {
  id: number;
  miniCurriculo: string;
  // user: UserUpdate;
  redesSociais: RedeSocial[];
  palestrantesEventos: Evento[];
}
